examples/QIR/Optimization/README.md

```diff
@@ -0,0 +1,274 @@
+
+## Prerequisites
+
+QIR generation is handled by the Q# compiler, while optimizations are performed at the LLVM level.
+The following software will be used in this walk-through:
+
+* Quantum Development Kit (QDK), containing the Q# compiler 
+* Clang, an LLVM based compiler for the C language family
+* LLVM optimizer, a tool to run custom optimization pipelines on LLVM IR code
+
+### Installing the QDK
+
+The *Quantum Development Kit (QDK)* documentation provides detailed guides on [installing the QDK](https://docs.microsoft.com/en-us/azure/quantum/install-overview-qdk#install-the-qdk-for-quantum-computing-locally) for various setups.
+This document will assume a command line setup for standalone Q# applications.
+
+Steps:
+
+* Install the [.NET Core SDK 3.1](https://dotnet.microsoft.com/download) (NOTE: use `dotnet-sdk-3.1` instead of `dotnet-sdk-5.0` in the Linux guides)
+* Install the QDK with `dotnet new -i Microsoft.Quantum.ProjectTemplates`
+* (Linux only) Ensure the [GNU OpenMP support library](https://gcc.gnu.org/onlinedocs/libgomp/) is installed on your system, e.g. via `sudo apt install libgomp1`
+
+### Installing Clang
+
+Depending on your platform, some LLVM tools may only be available by compiling from source.
+*Clang* however should be readily available from package managers or as pre-compiled binaries on most systems.
+LLVM's [official release page](https://releases.llvm.org/download.html) provides sources and binaries of LLVM and Clang.
+It's recommended to use LLVM version 11.1.0, as this is the version used by the Q# compiler. 
+
+Here are some suggestions on how to obtain Clang for different platforms:
+
+* **Ubuntu** : `sudo apt install clang-11` (NOTE: this installs the command `clang-11` not `clang`)
+* **Windows** : `choco install llvm --version=11.1.0` using [chocolatey](https://chocolatey.org/)
+* **macOS** : `brew install llvm@11` using [homebrew](https://brew.sh/)
+
+Pre-built binaries/installers:
+
+* **Ubuntu** : get `clang+llvm-11.1.0-x86_64-linux-gnu-ubuntu-20.10.tar.xz` from the [GitHub release](https://github.com/llvm/llvm-project/releases/tag/llvmorg-11.1.0)
+* **Windows** : get `LLVM-11.1.0-win64.exe` from the [GitHub release](https://github.com/llvm/llvm-project/releases/tag/llvmorg-11.1.0)
+* **macOS** : get `clang+llvm-11.0.0-x86_64-apple-darwin.tar.xz` from the [11.0.0 release](https://github.com/llvm/llvm-project/releases/tag/llvmorg-11.0.0) (11.1.0 not released)
+
+### Installing the LLVM optimizer
+
+If you followed the instructions above for Linux or macOS, your LLVM installation will have included the *LLVM optimizer*.
+You can test this by typing `opt` or `opt-11` in your terminal.
+
+Unfortunately, `opt` is not part of the official Windows distribution, which means you will to need to [compile LLVM from source](https://llvm.org/docs/GettingStarted.html) to get it.
```

Comment:
`opt` for Windows is provided by the `llvm-tools` and `llvmlib` `conda` packages.

Reply:
This is really useful, thank you!